File: python/day02/day02_test.py
from day02 import solve_part1
from day02 import solve_part2


def test_part1_with_empty_data():
    given = []
    expected = 0
    actual = solve_part1(given)

    assert expected == actual


def test_part1_with_example_data():
    given = ['forward 5', 'down 5', 'forward 8',
             'up 3', 'down 8', 'forward 2']
    expected = 150
    actual = solve_part1(given)

    assert expected == actual


# Part 2

def test_part2_with_empty_data():
    given = []
    expected = 0
    actual = solve_part2(given)

    assert expected == actual


def test_part2_with_example_data():
    given = ['forward 5', 'down 5', 'forward 8',
             'up 3', 'down 8', 'forward 2']
    expected = 900
    actual = solve_part2(given)

    assert expected == actual

File: java/adventofcode/src/test/java/com/github/mmacdonnacha/day02/Day02Test.java
package com.github.mmacdonnacha.day02;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;


public class Day02Test {

    private Day02 day02;

    @BeforeEach
    public void setup(){
        day02 = new Day02();
    }


    @Test
    public void emptyInputPart1(){
        List<String> data = new ArrayList<>();

        int expected = 0;
        int actual = day02.solvePart1(data);

        assertEquals(expected, actual);
    }


    @Test
    public void exampleInputPart1(){
        List<String> data = new ArrayList<>();
        data.add("forward 5");
        data.add("down 5");
        data.add("forward 8");
        data.add("up 3");
        data.add("down 8");
        data.add("forward 2");

        int expected = 150;
        int actual = day02.solvePart1(data);

        assertEquals(expected, actual);
    }


    /// Part 2
    
    @Test
    public void emptyInputPart2(){
        List<String> data = new ArrayList<>();

        int expected = 0;
        int actual = day02.solvePart2(data);

        assertEquals(expected, actual);
    }


    @Test
    public void exampleInputPart2(){
        List<String> data = new ArrayList<>();
        data.add("forward 5");
        data.add("down 5");
        data.add("forward 8");
        data.add("up 3");
        data.add("down 8");
        data.add("forward 2");

        int expected = 900;
        int actual = day02.solvePart2(data);

        assertEquals(expected, actual);
    }


}
